STANDARDS.md

```diff
@@ -1,3 +1,1269 @@
+also add `-Wmissing-export-lists` and `-Wmissing-deriving-strategies`: the 
+first ensures that we clearly indicate what is, and isn't, part of a module's 
+public API, and the second ensures that we have clarity about how everything 
+is derived. As we mandate both export lists and deriving strategies in this 
+document, these warnings ensure compliance, as well as checking it 
+automatically.
+
+### Linting
+
+Every source file MUST be free of warnings as produced by
+[HLint](http://hackage.haskell.org/package/hlint). The CI for any project MUST
+enforce this. 
+
+HLint warnings MUST be disabled by annotation use on a per-module basis when
+enforcing the recommendation would cause the code to no longer compile.
+
+#### Justification
+
+HLint automates away the detection of many common sources of boilerplate and 
+inefficiency. It also describes many useful refactors, which in many cases 
+make the code easier to read and understand. As this is fully automatic, it 
+saves effort on our part, and ensures consistency across the codebase without 
+us having to think about it.
+
+There are, however, some cases where HLint isn't as aware of GHC features and
+their interactions as we would like. One common example is that sometimes, full
+eta-expansion is required for GHC's type checker to accept some code, but HLint
+will argue for a reduction instead. In this case, disabling HLint warnings using
+annotations is the right choice, but this would usually only be in a narrow
+scope, and hence doesn't justify configuration changes project-wide. This also
+acts as a good signpost, as this behaviour can be unexpected for Haskell
+developers as well. 
+
+### Code formatting
+
+Every Haskell source file MUST be formatted according to
+[Fourmolu](http://hackage.haskell.org/package/fourmolu), with the following
+settings (as per its settings file):
+
+* `indentation: 2`
+* `comma-style: leading`
+* `record-brace-space: true`
+* `indent-wheres: true`
+* `diff-friendly-import-export: true`
+* `respectful: true`
+* `haddock-style: multi-line`
+* `newlines-between-decls: 1`
+
+Each source code line MUST be at most 80 characters wide.
+
+The project's Cabal file MUST be formatted according to
+[`cabal-fmt`](https://hackage.haskell.org/package/cabal-fmt).
+
+#### Justification
+
+Consistency is the most important part of readable codebases. Having a single,
+automatically-enforced, standard means that we can be sure everything will look
+similar, and not have to spend time or mind-space ensuring that our code
+complies with said standard. This also helps with `git diff`s, as the difference
+'spreads around' less.
+
+Lines wider than 80 characters become difficult to read, especially when viewed
+on a split screen. An 80-character maximum is also a long-standing convention,
+both within Haskell and also elsewhere. Lastly, very long lines tend to mean
+that we need better naming or refactoring.
```

Comment:
IMO we should leave the formatter & its config up to (sensible) choice. 
we were actually just considering switching to `stylish-haskell`, since it can automate the max char width

Reply:
`stylish-haskell` doesn't format a lot of things, unfortunately, which leaves too much style variation for my liking.

By 'leaving the formatter and its config up to (sensible) choice', what do you mean? Use only its defaults?